tests/api/admin-notes.php

```diff
@@ -314,5 +316,100 @@ public function test_get_notes_schema() {
 		$this->assertArrayHasKey( 'date_reminder_gmt', $properties );
 		$this->assertArrayHasKey( 'actions', $properties );
 		$this->assertArrayHasKey( 'is_snoozable', $properties );
+		$this->assertArrayHasKey( 'is_deleted', $properties );
+	}
+
+	/**
+	 * Test deleting a single note.
+	 */
+	public function test_delete_single_note() {
+		wp_set_current_user( $this->user );
+
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', $this->endpoint . '/3' ) );
+		$note     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( false, $note['is_deleted'] );
+
+		$response = $this->server->dispatch( new WP_REST_Request( 'DELETE', $this->endpoint . '/delete/3' ) );
+		$note     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( true, $note['is_deleted'] );
+	}
+
+	/**
+	 * Test undiong a single note delete.
+	 */
+	public function test_undo_single_notes_delete() {
+		wp_set_current_user( $this->user );
+
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', $this->endpoint . '/3' ) );
+		$note     = $response->get_data();
+
+		if ( $response->get_status() === 200 ) {
+			$response = $this->server->dispatch( new WP_REST_Request( 'DELETE', $this->endpoint . '/delete/3' ) );
+			$note     = $response->get_data();
+			$this->assertEquals( true, $note['is_deleted'] );
+		} else {
+			$this->assertEquals( 404, $response->get_status() );
+		}
+
+		$request = new WP_REST_Request( 'PUT', $this->endpoint . '/3' );
+		$request->set_body_params(
+			array(
+				'is_deleted' => '0',
+			)
+		);
+
+		$response = $this->server->dispatch( $request );
+		$note     = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( false, $note['is_deleted'] );
+	}
+
+	/**
+	 * Test deleting all the notes.
+	 */
+	public function test_delete_all_notes() {
+		wp_set_current_user( $this->user );
+
+		// It deletes only unactioned notes.
+		$response = $this->server->dispatch( new WP_REST_Request( 'DELETE', $this->endpoint . '/delete/all' ) );
+		$notes    = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 2, count( $notes ) );
+	}
+
+	/**
+	 * Test undiong all the notes delete.
+	 */
+	public function test_undo_all_notes_delete() {
+		wp_set_current_user( $this->user );
+
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', $this->endpoint ) );
+
+		if ( $response->get_status() === 200 ) {
+			$response = $this->server->dispatch( new WP_REST_Request( 'DELETE', $this->endpoint . '/delete/all' ) );
+			$notes    = $response->get_data();
+			$this->assertEquals( 2, count( $notes ) );
+		} else {
```

Comment:
How does this work, in a test? How do you get a 404?

Reply:
You are completely right, that control is not necessary here.
I removed it in the commit [8b183e9](https://github.com/woocommerce/woocommerce-admin/pull/4386/commits/8b183e9e52698fdf831077fbc9e5a33bc231d3c5) in the [PR 4386](https://github.com/woocommerce/woocommerce-admin/pull/4386)
Thank you 🙂